packages/expo-cli/src/projects.ts

```diff
@@ -50,4 +51,25 @@ async function ensureProjectExistsAsync(
   }
 }
 
-export { ensureProjectExistsAsync };
+/**
+ * Get the account and project name using a user and Expo config.
+ * This will validate if the owner field is set when using a robot account.
+ */
+function getProjectData(
+  user: User | RobotUser,
+  exp: ExpoConfig
+): Pick<ProjectData, 'accountName' | 'projectName'> {
+  if (user.kind === 'robot' && !exp.owner) {
+    throw new CommandError(
+      'ROBOT_OWNER_ERROR',
+      'The "owner" manifest property is required when using robot users.'
```

Comment:
Maybe add a link to https://docs.expo.io/versions/latest/config/app/#owner?

Reply:
Done!